bee-common/bee-common/src/packable.rs

```diff
@@ -38,6 +39,70 @@ pub trait Packable {
         Self: Sized;
 }
 
+impl Packable for bool {
+    type Error = std::io::Error;
+
+    fn packed_len(&self) -> usize {
+        (*self as u8).packed_len()
+    }
+
+    fn pack<W: Write>(&self, writer: &mut W) -> Result<(), Self::Error> {
+        (*self as u8).pack(writer)
+    }
+
+    fn unpack<R: Read + ?Sized>(reader: &mut R) -> Result<Self, Self::Error>
+    where
+        Self: Sized,
+    {
+        Ok(!matches!(u8::unpack(reader)?, 0))
+    }
+}
+
+/// Error that occurs on `Option<P: Packable>` operations.
+#[derive(Debug)]
+pub enum OptionError<P> {
```

Comment:
might be nitpicky, but instead of `P` I'ld name the generic parameter `E` here to indicate/insinuate that we use this type with some error type.

Reply:
Done